examples/keygen.rs

```diff
@@ -0,0 +1,425 @@
+//! Demonstrates how to use Nova to produce a recursive proof of the correct execution of
+//! iterations of the key generation function.
```

Comment:
I had a basic question about this example: What exactly is this proving? It seems like it is proving the knowledge of a set of secret keys? Does it make sense to turn this into proving knowledge of (one or more) secret keys such that their public keys are *in* a set of known public keys? Without such a restriction, it seems like the prover can just make up a set of secret keys and prove their knowledge, right?

Reply:
The use case for this example was taken from [Proof of private key ownership](https://medium.com/zokrates/building-identity-linked-zksnarks-with-zokrates-a36085cdd40#:~:text=the%20zkSNARK!-,Proof%20of%20Private%20Key%20Ownership,-In%20the%20context). Are you describing a membership proof? Changing the example to a better one is fine with me.

Reply:
Yes, the example is fine. But is the example making the claimed public key a public output, so the verifier can check if it is the same as some designated/well-known public key? If not, the proven statement isn't very useful, right? Because proving the knowledge of *some* secret key that corresponds to *some* public key isn't hard, right?

Reply:
From the web page the example is a proof of identity. The public key is a public output that the verifier would check against a public key that could be retrieved from a DID document for example. It may be the case that the example in isolation is not useful. The referenced web page for example uses two proofs for an identity-linked proof of knowledge.

- we have one function which proves knowledge of a secret
- we have another function which checks that a given public key was derived from a given private key